spec.html

```diff
@@ -43635,6 +43635,7 @@ <h1>Promise Objects</h1>
     </ul>
     <p>A promise is said to be <em>settled</em> if it is not pending, i.e. if it is either fulfilled or rejected.</p>
     <p>A promise is <em>resolved</em> if it is settled or if it has been &ldquo;locked in&rdquo; to match the state of another promise. Attempting to resolve or reject a resolved promise has no effect. A promise is <em>unresolved</em> if it is not resolved. An unresolved promise is always in the pending state. A resolved promise may be pending, fulfilled or rejected.</p>
+    <p>A <em>thenable</em> is an object `o` with `o.then(f, r)` method defined. Any Promise is thenable.</p>
```

Comment:
> Any Promise is thenable.

```js
const p = Promise.resolve();
p.__proto__ = null;
```
Is `p` thenable? No. Is `p` promise? For the spec - yes.

Reply:
```suggestion
    <p>A <em>thenable</em> is an object with a callable `then` property.</p>
```

Reply:
@zloirock 
Interesting example! I'd like to discuss your seemingly simple question "Is p promise?" a little bit more in a separate comment below and not to bury it in a "resolved conversation backyard".

@nicolo-ribaudo 
For consistency, my wording for the definition is modeled upon the wording just couple of paragraphs above it:

> A promise `p` is fulfilled if `p.then(f, r)` will immediately enqueue a Job to call the function `f`.

However, I'm fine with your wording as well. I'm not sure who is considered a maintainer of this repo to give an official approval to either wording and merge.

Reply:
BTW, the wording

> Any Promise is thenable.

can be adjusted to

> A promise created by the Promise Constructor is *thenable* upon creation.

What do you think?

Reply:
```js
delete `Promise.prototype.then`
```

Reply:
I understand your approach. I think that breaking the `[[Prototype]]` and overriding built-in properties breaks the spec for any type, not just `Promise`. You can do the same with `Number`. 

We could add "in unmodified standard environment" footnote for every second sentence in the spec to make it true even under the breaking runtime changes. I think that this is implied, but I don't see a point to continue this discussion here. It's better to merge something than nothing. I like @nicolo-ribaudo's suggestion.